src/Modal/Modal.js

```diff
@@ -1,50 +1,70 @@
-import React from 'react';
-import PropTypes from 'prop-types'
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Loading from '../Loading/Loading';
+import Error from '../Error/Error';
 import { getMovie } from '../util.js';
-import { Link } from 'react-router-dom'
+import { Link } from 'react-router-dom';
 import './Modal.scss';
 
 class Modal extends Component {
   constructor() {
     super();
     this.state = {
-      foundMovie: null
+      foundMovie: null,
+      loading: true,
+      errorThrown: false
     }
   }
 
+  throwError() {
+    this.setState({errorThrown: true, loading: false})
+  }
+
   componentDidMount() {
     getMovie(this.props.id)
-    .then(movie => this.setState({foundMovie: movie.movie}))
-    .catch(error => console.log(error))
+    .then(movie => {
+      return this.setState({foundMovie: movie.movie, loading: false})
+    })
+    .catch(error => {
+      console.log(error)
+      this.throwError()
+    })
   }
 
   render() {
     const foundMovie = this.state.foundMovie
-    return (
-      (!foundMovie) ? <p>No movie clicked yet</p> : 
-      <section className="modal">
-        <img src={foundMovie.backdrop_path} alt="movie backdrop"/>
-        <div className="m-data">
-          <p className="m-tagline">{foundMovie.tagline}</p>
-          <h2 className="m-title">{foundMovie.title}</h2>
-          <p className="m-date">Release Date: {foundMovie.release_date}</p>
-          <p className="m-rating">Average Rating: {foundMovie.average_rating.toFixed(1)}/10.0</p>
-          <p className="m-runtime">Runtime: {foundMovie.runtime} minutes</p>
-          <p className="m-overview">{foundMovie.overview}</p>
-          <div className="m-genre">
-            <p>Filed under:</p>
-              <ul>{listGenres}</ul>
+    if (this.state.errorThrown) {
+      return <Error />
+    } else if (this.state.loading) {
+      return <Loading />
+    } else {
+      return (
+        (foundMovie === null) ? <main className="loading">HEY</main> : 
+        <section className="modal">
+          <img src={foundMovie.backdrop_path} alt="movie backdrop"/>
+          <div className="m-data">
+            <p className="m-tagline">{foundMovie.tagline}</p>
+            <h2 className="m-title">{foundMovie.title}</h2>
+            <p className="m-date">Release Date: {foundMovie.release_date}</p>
+            <p className="m-rating">Average Rating: {foundMovie.average_rating.toFixed(1)}/10.0</p>
+            <p className="m-runtime">Runtime: {foundMovie.runtime} minutes</p>
+            <p className="m-overview">{foundMovie.overview}</p>
+            <div className="m-genre">
+              <p>Filed under:</p>
+                <ul>{foundMovie.genres.map(genre => <li key={genre}>{genre}</li>)}</ul>
```

Comment:
Technically this counts as rendering a list, so React wanted to make sure it had unique keys.